LEGO1/omni/src/action/mxdsstreamingaction.cpp

```diff
@@ -5,94 +5,97 @@
 DECOMP_SIZE_ASSERT(MxDSStreamingAction, 0xb4)
 
 // FUNCTION: LEGO1 0x100cd010
+// FUNCTION: BETA10 0x1015f380
 MxDSStreamingAction::MxDSStreamingAction(MxDSAction& p_dsAction, MxU32 p_offset)
 {
 	Init();
 
-	*this = p_dsAction;
-	this->m_unk0x94 = p_offset;
-	this->m_bufferOffset = p_offset;
+	*(MxDSAction*) this = p_dsAction;
+	m_unk0x94 = p_offset;
+	m_bufferOffset = p_offset;
 }
 
 // FUNCTION: LEGO1 0x100cd090
+// FUNCTION: BETA10 0x101565a0
 MxBool MxDSStreamingAction::HasId(MxU32 p_objectId)
 {
-	if (this->m_internalAction) {
-		return this->m_internalAction->HasId(p_objectId);
-	}
-	return FALSE;
+	return m_internalAction ? m_internalAction->HasId(p_objectId) : FALSE;
 }
 
 // FUNCTION: LEGO1 0x100cd0d0
+// FUNCTION: BETA10 0x101564a0
 MxDSStreamingAction::MxDSStreamingAction(MxDSStreamingAction& p_dsStreamingAction)
 {
 	Init();
 	CopyFrom(p_dsStreamingAction);
 }
 
 // FUNCTION: LEGO1 0x100cd150
+// FUNCTION: BETA10 0x1015f41d
 MxDSStreamingAction::~MxDSStreamingAction()
 {
-	if (this->m_unk0xa0) {
-		delete this->m_unk0xa0;
+	if (m_unk0xa0) {
+		delete m_unk0xa0;
 	}
-	if (this->m_unk0xa4) {
-		delete this->m_unk0xa4;
+	if (m_unk0xa4) {
+		delete m_unk0xa4;
 	}
-	if (this->m_internalAction) {
-		delete this->m_internalAction;
+	if (m_internalAction) {
+		delete m_internalAction;
 	}
 }
 
 // FUNCTION: LEGO1 0x100cd1e0
-MxResult MxDSStreamingAction::Init()
+// FUNCTION: BETA10 0x1015f53c
+void MxDSStreamingAction::Init()
 {
-	this->m_unk0x94 = 0;
-	this->m_bufferOffset = 0;
-	this->m_unk0x9c = 0;
-	this->m_unk0xa0 = NULL;
-	this->m_unk0xa4 = NULL;
-	this->m_unk0xa8 = 0;
-	this->m_unk0xac = 2;
-	this->m_internalAction = NULL;
-	return SUCCESS;
+	m_unk0x94 = 0;
+	m_bufferOffset = 0;
+	m_unk0x9c = 0;
+	m_unk0xa0 = NULL;
+	m_unk0xa4 = NULL;
+	m_unk0xa8 = 0;
+	m_unk0xac = 2;
+	m_internalAction = NULL;
 }
 
 // FUNCTION: LEGO1 0x100cd220
+// FUNCTION: BETA10 0x1015f5b9
 MxDSStreamingAction* MxDSStreamingAction::CopyFrom(MxDSStreamingAction& p_dsStreamingAction)
 {
-	*this = p_dsStreamingAction;
-	this->m_unk0x94 = p_dsStreamingAction.m_unk0x94;
-	this->m_bufferOffset = p_dsStreamingAction.m_bufferOffset;
-	this->m_unk0x9c = p_dsStreamingAction.m_unk0x9c;
-	this->m_unk0xa0 = NULL;
-	this->m_unk0xa4 = NULL;
-	this->m_unk0xac = p_dsStreamingAction.m_unk0xac;
-	this->m_unk0xa8 = p_dsStreamingAction.m_unk0xa8;
+	*(MxDSAction*) this = p_dsStreamingAction;
```

Comment:
I think the `MxDSAction::operator=(p_dsStreamingAction);` would be cleaner and I think we have used that in other places too